packages/cdk/src/config.ts

```diff
@@ -0,0 +1,214 @@
+import { GetParametersByPathCommand, SSMClient } from '@aws-sdk/client-ssm';
+import {
+  ExternalSecret,
+  ExternalSecretPrimitive,
+  ExternalSecretPrimitiveType,
+  MedplumInfraConfig,
+  MedplumSourceInfraConfig,
+  OperationOutcomeError,
+  badRequest,
+  validationError,
+} from '@medplum/core';
+
+const DEFAULT_AWS_REGION = 'us-east-1';
+const VALID_PRIMITIVE_TYPES = ['string', 'boolean', 'number'];
+
+export async function fetchParameterStoreSecret(path: string, key: string): Promise<string> {
+  const region = DEFAULT_AWS_REGION;
+  const client = new SSMClient({ region });
+
+  let nextToken: string | undefined;
+  do {
+    const response = await client.send(
+      new GetParametersByPathCommand({
+        Path: path,
+        NextToken: nextToken,
+        WithDecryption: true,
+      })
+    );
+    if (response.Parameters) {
+      for (const param of response.Parameters) {
+        const paramKey = (param.Name as string).replace(path, '');
+        if (paramKey === key) {
+          if (!param.Value) {
+            throw new OperationOutcomeError(badRequest(`Key '${key}' found in path '${path}' but missing a value.`));
+          }
+          return param.Value;
+        }
+      }
+    }
+    nextToken = response.NextToken;
+  } while (nextToken);
+
+  throw new OperationOutcomeError(
+    badRequest(
+      `Key '${key}' not found at path '${path}'. Make sure your path and key are correct and are defined in your Parameter Store.`
+    )
+  );
+}
+
+export function normalizeFetchedValue(
+  key: string,
+  rawValue: ExternalSecretPrimitive,
+  expectedType: ExternalSecretPrimitiveType
+): ExternalSecretPrimitive {
+  const typeOfVal = typeof rawValue;
+  // Return raw type if type is string and value is of type string, or if type isn't string and typeof val isn't string
+  if (!VALID_PRIMITIVE_TYPES.includes(typeOfVal)) {
+    throw new OperationOutcomeError(
+      validationError(
+        `Invalid value found for type, expected 'string' or 'boolean' or 'number', received: '${typeOfVal}'`
+      )
+    );
+  }
+  if (expectedType === 'string' && typeOfVal === 'string') {
```

Comment:
I think we can reduce the number of error checks if we boil this down to 3 possibilities:
1. `typeOfVal === expectedType`, so return as-is
2. `typeOfVal === 'string'`, so convert it
3. Otherwise it's an error